fitbenchmarking/results_processing/base_table.py

```diff
@@ -518,7 +518,7 @@ def vals_to_colour(vals, cmap, cmap_range, log_ulim):
 
         return hex_strs
 
-    def save_colourbar(self, fig_dir, n_divs=100, sz_in=(3, 0.8)):
+    def save_colourbar(self, fig_dir, n_divs=100, sz_in=(3, 0.8)) -> str:
```

Comment:
Have you used type annotations much? We haven't used them on this project yet but there's no reason not to.

Reply:
Yes I tend to use them all the time, when I remember to. They've been quite useful to me in the past, especially when coming across a new piece of code. Other than being good for documentation, there are other benefits too https://stavshamir.github.io/python/the-other-benefit-of-python-type-annotations/

I noticed that you specify the `type` of parameters in the docstring so maybe its not necessary to use type annotations in this case?

Reply:
I don't see a problem with them being here, we should probably use them more... I heard about "mypy" as well recently so that may be something to aim for if we start adding them everywhere